File: demo/src/main/java/com/guoxiaoxing/kotlin/demo/java/Person.java
package com.guoxiaoxing.kotlin.demo.java;

/**
 * For more information, you can visit https://github.com/guoxiaoxing or contact me by
 * guoxiaoxingse@163.com.
 *
 * @author guoxiaoxing
 * @since 2017/5/22 下午5:48
 */
public abstract class Person {

    private String name;
    private int age;

    public Person(String name){
        this.name = name;
    }

    public Person(String name, int age){
        this.name = name;
        this.age = age;
    }

    public void say(String message){
        System.out.print(message);
    }
}

File: demo/src/main/java/com/guoxiaoxing/kotlin/demo/oop/Person.kt
package com.guoxiaoxing.kotlin.demo.oop

/**
 * For more information, you can visit https://github.com/guoxiaoxing or contact me by
 * guoxiaoxingse@163.com.

 * @author guoxiaoxing
 * *
 * @since 2017/5/24 下午4:31
 */
open class Person {

    open val name : String = ""

    constructor(name: String)

    constructor(name: String, age: Int) : this(name)

    open fun say(){

        var data = DataClass("jack", 25)
        data.copy(age = 20)
    }


}
